### DISCUSSION TITLE
Query Regarding MP3 Format Support and Accuracy in Silero-VAD

### DISCUSSION THREAD

**Discussion by Jellun** [snakers4/silero-vad#395] [Q&A] [answered]
Hi @snakers4 ,

I've noticed that all your example code for silero-vad uses only WAV format. However, I tested the model using an MP3 file:

audio = read_audio("test.mp3", sampling_rate=SAMPLING_RATE)

The model operates correctly with MP3 files, but there appears to be a reduction in accuracy compared to WAV format. It's important to mention that the MP3 files I tested were high quality (i.e., not excessively compressed).

Could you clarify whether silero-vad inherently supports MP3 format? Also, is there a noticeable difference in quality or accuracy between MP3 and WAV formats?

Thanks in advance.
Jun

**Comment by snakers4** [comment]
> Could you clarify whether silero-vad inherently supports MP3 format? Also, is there a noticeable difference in quality or accuracy between MP3 and WAV formats?

To be honest - we did not check this.

We did not explicitly add MP3 (to and back) augmentations to the training process.
Typically most of source data is in WAV / OGG-VORBIS / OGG-OPUS.
We store data in OGG-OPUS. 
High quality MP3 files provide quite good quality, but we did not explicitly test for which bitrate there is which quality degradation.

**Reply by Jellun** [reply]
Does method **read_audio("audio_file_path", sampling_rate=SAMPLING_RATE)** use FFMPEG or some other Python modules to load the file? Does it automatically detect the audio file format and convert it to wav before further processing it within the model?

**Reply by snakers4** [reply, marked as answer]
> read_audio("audio_file_path", sampling_rate=SAMPLING_RATE) use FFMPEG or some other Python modules to load the file? Does it automatically detect the audio file format and convert it to wav before further processing it within the model?

I believe it uses torchaudio, which has a `sox_io` backend now by default, which uses sox. Not sure about MP3 support in sox, it is always flaky with MP3.
